pkg/sidecar/sidecar_test.go

```diff
@@ -21,9 +22,285 @@ import (
+				// where the sidecar might try to write to the channels
+				// and panic before the test as a whole aborts.
+				require.NoError(t, ctx.Err())
+				return
+			case processState = <-s.cmdExitChan:
+				exited = true
+				t.Logf("Command exited with %s", processState.String())
+			}
+
+			// If we expect the process to exit, ensure it has
+			require.Equal(t, tc.expectTerminated, exited)
+
+			// We only check the exit status if the process is
+			// supposed to exit; some tests will leave the process
+			// running.
+			if tc.expectTerminated {
+				// Sidecar monitor must agree it has exited
+				require.Equal(t, false, s.sidecar.processRunning)
+				if tc.expectSignalExit > 0 {
+					// Does this need to be in a separate
+					// abstraction with a build guard for
+					// windows, or is it sufficient that it
+					// is unreachable on Windows?
+					require.Equal(t, tc.expectSignalExit, processState.Sys().(syscall.WaitStatus).Signal())
+				} else {
+					require.True(t, processState.Exited())
+					require.Equal(t, tc.expectExitStatus, processState.ExitCode())
+				}
+			} else {
+				// The tests require that the process is still
+				// running. Make sure you allow enough time for
+				// the launched process to keep running before
+				// the test ends.
+				require.Equal(t, true, s.sidecar.processRunning)
+			}
+
+			// The test file must have been created if expected
+			if tc.expectFileExists != "" {
+				_, err := os.Stat(tc.expectFileExists)
+				require.NoError(t, err)
+			}
+
+			// Consume and log any stdout/stderr. There's no test facility
+			// for making assertions about it yet, or passing stdin.
+			if tc.logStdOutErr {
+				t.Logf("stdout/stderr: %s", s.sidecar.stdout.(*strings.Builder).String())
+			}
+
+			// If a process was left running the sidecar will still
+			// be waiting for it to exit. When it does, it will
+			// write to the cmdExitChan. We don't really want to
+			// wait for the process to finish so we'll orphan
+			// channels in this case. We could start goroutines to
+			// read from them then close them but why bother, the
+			// whole test will be terminated soon and the GC will
+			// take care of it once the Sidecar is done anyway.
+			if exited {
+				s.Close(t)
+			}
+		})
+	}
+}
+
+// A short-lived process gets re-launched whenever the certs are rotated. The
+// pid must change in each iteration.
+//
+// This test should probably also assert that the pid doesn't change between
+// cert rotations, but that's a bit more complex to test.
+func TestSidecar_TestCmdRunsRelaunchShortlived(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	s := newSidecarTest(t)
+	defer s.Close(t)
+
+	config := s.sidecar.config
+	if runtime.GOOS == "windows" {
+		config.Cmd = "cmd"
+		config.CmdArgs = "/c echo hello"
+	} else {
+		config.Cmd = "echo"
+		config.CmdArgs = "hello"
+	}
+
+	var pid int
+	for range 3 {
```

Comment:
Is it necessary to repeat this 3 times?

Reply:
3 is arbitrary, it's just to ensure that re-issuing works.